### PR TITLE
Base Repentogon Support

### PR DESCRIPTION
**Pull request by Syntax-Sculptor**
This only includes new enums and interfaces. A separate repo will be published for the new functions which modify existing classes upon merge.

### CODE REVIEW

File: packages/isaacscript-spell/dictionaries/isaac/api.txt [hunk 1]
```diff
@@ -1,6 +1,5 @@
-# These are words from the Lua API. Words from misspelled enums are not included, since those are
 # fixed in the reimplemented IsaacScript enums.
-
+# These are words from the Lua API. Words from misspelled enums are not included, since those are
```
Comment (Zamiell) [inline packages/isaacscript-spell/dictionaries/isaac/api.txt:2]: what are these changes?

**Comment by Zamiell** [comment]
are the enums for repentegon only? if so, then adding them next to vanilla enums is the wrong place, right?

**Comment by Syntax-Sculptor** [comment]
> this is too big for a single PR, can we break it up into multiple PRs?

Sure.

> are the enums for repentegon only? if so, then adding them next to vanilla enums is the wrong place, right?

Are you talking about them being in the same directory?

**Comment by Zamiell** [comment]
like for example AnnouncerVoiceMode

**Comment by Syntax-Sculptor** [comment]
> like for example AnnouncerVoiceMode

In that case, no. Those enums can be used outside of Repentogon (i.e: the AnnouncerVoiceMode enum would be useful even without Repentogon since there's `Options.AnnouncerVoiceMode`)

**Comment by Zamiell** [comment]
right but the PR doesn't actually change the type of `AnnouncerVoiceMode`, so thats a bug then?

**Comment by Zamiell** [comment]
why dont we start with just a PR to the options and option related enums.
i dont see why that should be inside of a repentagon PR

**Pull request closed without merging**